src/DropDownMenu/DropDownMenu.jsx

```diff
@@ -0,0 +1,444 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import Transitions from '../styles/transitions';
+import DropDownArrow from '../svg-icons/navigation/arrow-drop-down';
+import Menu from '../menus/menu';
+import MenuItem from '../menus/menu-item';
+import DefaultRawTheme from '../styles/raw-themes/light-raw-theme';
+import ClearFix from '../clearfix';
+import ThemeManager from '../styles/theme-manager';
+import Popover from '../popover/popover';
+import PopoverAnimationFromTop from '../popover/popover-animation-from-top';
+import styleUtils from '../utils/styles';
+import warning from 'warning';
+
+const DropDownMenu = React.createClass({
+
+  contextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  //for passing default theme context to children
+  childContextTypes: {
+    muiTheme: React.PropTypes.object,
+  },
+
+  getChildContext() {
+    return {
+      muiTheme: this.state.muiTheme,
+    };
+  },
+
+  // The nested styles for drop-down-menu are modified by toolbar and possibly
+  // other user components, so it will give full access to its js styles rather
+  // than just the parent.
+  propTypes: {
+    /**
+     * The width will automatically be set according to the items inside the menu.
+     * To control this width in css instead, set this prop to false.
+     */
+    autoWidth: React.PropTypes.bool,
+
+    /**
+     * The `MenuItem`s to populate the `Menu` with. If the `MenuItems` have the
+     * prop `label` that value will be used to render the representation of that
+     * item within the field.
+     */
+    children: React.PropTypes.node,
+
+    /**
+     * The css class name of the root element.
+     */
+    className: React.PropTypes.string,
+
+    /**
+     * Disables the menu.
+     */
+    disabled: React.PropTypes.bool,
+
+    /**
+     * **DEPRECATED** `DropDownMenu` will use this member to display
+     * the name of the item.
+     */
+    displayMember: React.PropTypes.string,
+
+    /**
+     * Overrides the styles of icon element.
+     */
+    iconStyle: React.PropTypes.object,
+
+    /**
+     * **DEPRECATED** `DropDownMenu` will use this member to display
+     * the name of the item on the label.
+     */
+    labelMember: React.PropTypes.string,
```

Comment:
> I want to improve the doc generation in 2-3 ways. for functions, deprecated, and categorization.

I agree, there is room for improvement. For instance by using the `deprecatedProp` we could add more than just `**DEPRECATED**` but also what to do instead